packages/client/src/components/form/FormFieldGenerator.tsx

```diff
@@ -869,11 +914,47 @@ class FormSectionComponent extends React.Component<Props> {
     })
   }
 
+  getOnClickHandle = (
+    fields: IFormField[],
+    field: IButtonFormField,
+    setFieldValue: (fieldName: string, value: any) => void,
+    setFieldError: (fieldName: string, message: string) => void
+  ) => {
+    const trigger = fields.find((f) => f.name === field.options.trigger)
+    if (trigger && isFieldHttp(trigger)) {
+      return () => {
+        const {
+          options: { url, ...requestOptions }
+        } = trigger
+        const fieldsDependentOnTrigger = getDependentFields(fields, trigger)
+        fetch(url, requestOptions)
+          .then((res) => res.text())
+          .then((data) => {
+            setFieldValue(`${field.name}.data`, data)
+            if (fieldsDependentOnTrigger.length) {
+              fieldsDependentOnTrigger.forEach((field) => {
+                setFieldValue(field.name, data)
```

Comment:
https://github.com/opencrvs/opencrvs-core/pull/7490/files#diff-0de97bf0a8a09efc3b474e18a771ce127696c7817508de4a2ca557248d846c53R36

It's better to have some type of a `{data, error, loading}` object here that you emit so other inputs can always rely the data structure of this input's value to be the same regardless of its current state

Reply:
Thanks, @rikukissa. I am on  it